src/Symfony/Component/Clock/MockClock.php

```diff
@@ -38,23 +43,29 @@ public function __construct(\DateTimeImmutable|string $now = 'now', \DateTimeZon
             }
         }
 
+        $this->timezone = $timezone;
+
         if (\is_string($now)) {
-            $now = new DatePoint($now, $timezone ?? new \DateTimeZone('UTC'));
-        } elseif (!$now instanceof DatePoint) {
+            $now = new DatePoint($now, $this->timezone ?? new \DateTimeZone('UTC'));
+        } elseif (!$now instanceof DatePoint && !\is_callable($now)) {
             $now = DatePoint::createFromInterface($now);
         }
 
-        $this->now = null !== $timezone ? $now->setTimezone($timezone) : $now;
+        if (\is_callable($now)) {
```

Comment:
You don't allow arbitrary callables, only closures.

```suggestion
        if ($now instanceof \Closure) {
```